### PR TITLE
refactor: session

### PR DESCRIPTION
**Pull request by prmukherj**
Refactoring the methods exposed in session level.

### CODE REVIEW

File: src/ansys/fluent/core/solver/flobject.py [hunk 1]
```diff
@@ -177,11 +177,11 @@ def __init__(self, name: Optional[str] = None, parent=None):
         if name is not None:
             self._setattr("_name", name)
 
-    def set_flproxy(self, flproxy):
+    def _set_flproxy(self, flproxy):
```
Comment (mkundu1) [inline src/ansys/fluent/core/solver/flobject.py:180]: This change will require some Fluent code change in the new PyConsole. I think, at this stage, we should avoid any PyFluent change that will require sync changes in Fluent. Please also test this PR with the new PyConsole in Fluent to catch any other issue.

Comment (prmukherj) [reply src/ansys/fluent/core/solver/flobject.py]: Checked this PR in PyConsole. Working fine.

Comment (prmukherj) [reply src/ansys/fluent/core/solver/flobject.py]: 'fluent_connection' and  'set_flproxy' will be handled as part of separate PRs.

File: src/ansys/fluent/core/fluent_connection.py [hunk 2]
```diff
@@ -265,7 +265,7 @@ def __init__(
         PortNotProvided
             If port is not provided.
         """
-        self.error_state = ErrorState()
+        self._error_state = ErrorState()
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/fluent_connection.py:268]: 👍

File: src/ansys/fluent/core/session.py [hunk 3]
```diff
@@ -205,6 +210,33 @@ def build_from_fluent_connection(
...
+        """Return the SolutionVariableData handle."""
+        warnings.warn(
+            "field_data is deprecated, use fields.field_data instead",
+            DeprecationWarning,
+        )
+        return self.fields.field_data
+
+    @property
+    def field_data_streaming(self):
+        """Return the SolutionVariableData handle."""
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/session.py:233]: Incorrect doc comment

File: src/ansys/fluent/core/session.py [hunk 4]
```diff
@@ -221,7 +253,7 @@ def stop_journal(self):
         self.journal.stop()
 
     @classmethod
-    def create_from_server_info_file(
+    def _create_from_server_info_file(
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/session.py:256]: :+1:

File: src/ansys/fluent/core/session_solver.py [hunk 5]
```diff
@@ -116,36 +116,54 @@ def _build_from_fluent_connection(self, fluent_connection):
...
-            self._solution_variable_service, self.solution_variable_info
+            self._solution_variable_service, self.fields.solution_variable_info
         )
 
     @property
     def svar_data(self):
         """Return the SolutionVariableData handle."""
         warnings.warn(
-            "svar_data is deprecated, use solution_variable_data instead",
+            "svar_data is deprecated, use fields.solution_variable_data instead",
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/session_solver.py:136]: We could presumably remove the warnings at the `svar_*` level now. We will get them automatically next level down.

File: src/ansys/fluent/core/session_solver.py [hunk 6]
```diff
@@ -246,29 +264,26 @@ def read_case_lightweight(self, file_name: str):
...
-        self._root.set_state(state, **kwargs)
+        self.root.set_state(state, **kwargs)
 
     def __call__(self):
         return self.get_state()
 
     def _populate_settings_api_root(self):
         if not self._settings_api_root:
-            self._settings_api_root = _import_settings_root(self._root)
+            self._settings_api_root = _import_settings_root(self.root)
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/session_solver.py:278]: 👍  Correctly made this public. The next level of thinking is that the name is actually a bit vague. How about `settings`?
```
>>> state = solver.settings.get_state()
>>> settings = solver.settings
>>> settings.setup.models...
```

File: src/ansys/fluent/core/session_solver.py [hunk 7]
```diff
@@ -246,29 +264,26 @@ def read_case_lightweight(self, file_name: str):
...
-        return sorted(
-            set(
-                list(self.__dict__.keys())
-                + dir(type(self))
-                + dir(self._settings_api_root)
-            )
-        )
+        dir_list = set(
+            list(self.__dict__.keys()) + dir(type(self)) + dir(self._settings_api_root)
+        ) - {"svar_data", "svar_info"}
```
[the remark this answers is not in the document]
Comment (prmukherj) [reply src/ansys/fluent/core/session_solver.py:288]: will do.

File: src/ansys/fluent/core/session_solver.py (hunk 6 above)
Comment (seanpearsonuk) [reply src/ansys/fluent/core/session_solver.py:278]: N.b. in the (new) console right now:
```
>>> import ansys.fluent.core.solver.flobject
>>> G_ = ansys.fluent.core.solver.flobject.Group
>>> [x for x in dir(solver) if isinstance(getattr(solver, x), G_)]  
['_root', '_settings_root', 'design', 'file', 'mesh', 'parallel', 'parameters', 'results', 'server', 'setup', 'solution']
>>> 
```
Both '_root' and '_settings_root' are in the list separately.  We only would like `settings` in both PyFluent & PyConsole.

@mkundu1 @prmukherj

File: src/ansys/fluent/core/session.py (hunk 3 above)
Comment (prmukherj) [reply src/ansys/fluent/core/session.py:233]: Done, Thank you.

File: src/ansys/fluent/core/session_solver.py (hunk 5 above)
Comment (prmukherj) [reply src/ansys/fluent/core/session_solver.py:136]: @seanpearsonuk, I will remove the warning from solution_variable_* for the reason mentioned.

File: src/ansys/fluent/core/session_solver.py (hunk 6 above)
Comment (mkundu1) [reply src/ansys/fluent/core/session_solver.py:278]: @seanpearsonuk Are we thinking to supersede `solver.setup` by `solver.settings.setup`? I agree that the grouping all top level settings object under a common head is better, but we need to plan the corresponding Python journal recording change in Fluent.

File: src/ansys/fluent/core/session_solver.py [hunk 8]
```diff
@@ -116,36 +116,54 @@ def _build_from_fluent_connection(self, fluent_connection):
...
         super(Solver, self).build_from_fluent_connection(fluent_connection)
         self._build_from_fluent_connection(fluent_connection)
 
-    @property
-    def solution_variable_data(self) -> SolutionVariableData:
+    def _solution_variable_data(self) -> SolutionVariableData:
         """Return the SolutionVariableData handle."""
         return service_creator("svar_data").create(
-            self._solution_variable_service, self.solution_variable_info
+            self._solution_variable_service, self.fields.solution_variable_info
```
[the remark this answers is not in the document]
Comment (seanpearsonuk) [reply src/ansys/fluent/core/session_solver.py:129]: N.b. agreed to put reductions here too

File: src/ansys/fluent/core/session_solver.py (hunk 6 above)
Comment (seanpearsonuk) [reply src/ansys/fluent/core/session_solver.py:279]: n.b. agreed to keep both formats in 242 intending to deprecate solver.setup etc in 251 and update the journaling

File: src/ansys/fluent/core/session.py [hunk 9]
```diff
@@ -205,6 +210,33 @@ def build_from_fluent_connection(
...
+        """Provides access to Fluent field information."""
+        warnings.warn(
+            "field_info is deprecated, use fields.field_info instead",
+            DeprecationWarning,
+        )
+        return self.fields.field_info
+
+    @property
+    def field_data(self):
+        """Provides access to Fluent field data on surfaces."""
```
Comment (PipKat) [inline src/ansys/fluent/core/session.py:224]:
```suggestion
        """Fluent field data on surfaces."""
```

File: src/ansys/fluent/core/session_solver.py [hunk 10]
```diff
@@ -116,36 +116,54 @@ def _build_from_fluent_connection(self, fluent_connection):
...
-            "svar_data is deprecated, use solution_variable_data instead",
+            "svar_data is deprecated, use fields.solution_variable_data instead",
             DeprecationWarning,
         )
-        return self.solution_variable_data
+        return self.fields.solution_variable_data
 
     @property
     def svar_info(self):
         """Return the SolutionVariableInfo handle."""
```
Comment (PipKat) [inline src/ansys/fluent/core/session_solver.py:143]:
```suggestion
        """SolutionVariableInfo handle."""
```
Other libraries surround code entities in double backticks. This doesn't seem to the case in this library.

File: src/ansys/fluent/core/session_solver.py [hunk 11]
```diff
@@ -116,36 +116,54 @@ def _build_from_fluent_connection(self, fluent_connection):
...
+            "svar_info is deprecated, use fields.solution_variable_info instead",
+            DeprecationWarning,
+        )
+        return self.fields.solution_variable_info
+
+    @property
+    def solution_variable_data(self):
+        """Return the SolutionVariableData handle."""
+        warnings.warn(
+            "solution_variable_data is deprecated, use fields.solution_variable_data instead",
```
Comment (PipKat) [inline src/ansys/fluent/core/session_solver.py:154]:
```suggestion
            "solution_variable_data is deprecated. Use fields.solution_variable_data instead.",
```

File: src/ansys/fluent/core/session.py [hunk 12]
```diff
@@ -205,6 +210,33 @@ def build_from_fluent_connection(
...
+            "field_data is deprecated, use fields.field_data instead",
+            DeprecationWarning,
+        )
+        return self.fields.field_data
+
+    @property
+    def field_data_streaming(self):
+        """Field gRPC streaming service of Fluent."""
+        warnings.warn(
+            "field_data_streaming is deprecated, use fields.field_data_streaming instead",
```
Comment (PipKat) [inline src/ansys/fluent/core/session.py:235]:
```suggestion
            "field_data_streaming is deprecated. Use fields.field_data_streaming instead.",
```

File: src/ansys/fluent/core/session_solver.py (hunk 5 above)
Comment (PipKat) [inline src/ansys/fluent/core/session_solver.py:136]:
```suggestion
            "svar_data is deprecated. Use fields.solution_variable_data instead.",
```

File: src/ansys/fluent/core/session_solver.py [hunk 13]
```diff
@@ -116,36 +116,54 @@ def _build_from_fluent_connection(self, fluent_connection):
...
             DeprecationWarning,
         )
-        return self.solution_variable_info
+        return self.fields.solution_variable_data
+
+    @property
+    def solution_variable_info(self):
+        """Return the SolutionVariableInfo handle."""
+        warnings.warn(
+            "solution_variable_info is deprecated, use fields.solution_variable_info instead",
```
Comment (PipKat) [inline src/ansys/fluent/core/session_solver.py:163]:
```suggestion
        """SolutionVariableInfo handle."""
        warnings.warn(
            "solution_variable_info is deprecated. Use fields.solution_variable_info instead.",
```

File: src/ansys/fluent/core/solver/flobject.py [hunk 14]
```diff
@@ -189,7 +189,7 @@ def set_flproxy(self, flproxy):
         """Set flproxy object."""
         self._setattr("_flproxy", flproxy)
 
-    def set_file_transfer_service(self, file_transfer_service):
+    def _set_file_transfer_service(self, file_transfer_service):
         """Set file_transfer_service."""
```
Comment (PipKat) [inline src/ansys/fluent/core/solver/flobject.py:193]:
```suggestion
        """Set the file transfer service."""
```
It is best practice to use plain English in descriptions whenever possible.

File: src/ansys/fluent/core/solver/flobject.py [hunk 15]
```diff
@@ -360,27 +360,27 @@ def after_execute(self, value):
         if hasattr(self, "_do_after_execute"):
             self._do_after_execute(value)
 
-    def while_setting_state(self):
+    def _while_setting_state(self):
         """Avoid additional processing while setting state."""
```
Comment (PipKat) [inline src/ansys/fluent/core/solver/flobject.py:364]:
```suggestion
        """Avoid additional processing while setting the state."""
```

**Review by mkundu1 — APPROVED** [review]

**Review by hpohekar — APPROVED** [review]

File: src/ansys/fluent/core/session_solver.py (hunk 10 above)
Comment (prmukherj) [reply src/ansys/fluent/core/session_solver.py]: Will do that.

File: src/ansys/fluent/core/solver/flobject.py [hunk 16]
```diff
@@ -360,27 +360,27 @@ def after_execute(self, value):
...
         return nullcontext()
 
-    def while_renaming(self):
+    def _while_renaming(self):
         """Avoid additional processing while renaming."""
```
[the remark this answers is not in the document]
Comment (prmukherj) [reply src/ansys/fluent/core/solver/flobject.py:384]: These are private methods now, does not have user exposed docs.

**Review by seanpearsonuk — APPROVED** [review]

File: src/ansys/fluent/core/session_solver.py [hunk 17]
```diff
@@ -93,7 +93,7 @@ def __init__(
     def _build_from_fluent_connection(self, fluent_connection):
         self._tui_service = self.datamodel_service_tui
         self._se_service = self.datamodel_service_se
-        self._settings_service = self.settings_service
+        self._settings_service = self._settings_service
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/session_solver.py:96]: @prmukherj This line looks redundant

**Pull request merged**